### PR TITLE
fix: crash on GrapheneOS when downloading certificate (WPB-7407)

### PR DESCRIPTION
**Pull request by alexandreferris**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-7407" title="WPB-7407" target="_blank">BugWPB-7407</a>  [Android] Crash on graphene OS when downloading own or another users certificate
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [X] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [X] contains a reference JIRA issue number like `SQPIT-764`
    - [X] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [X] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

There is 2 issues:
- When downloading certificate when user is in GrapheneOS it would crash the app
- When downloading certificate it would download twice (double files)

### Causes (Optional)

- Crash : we need to ask file writing permissions for users in GrapheneOS.
- double files : We generate the certificate file and was downloading the generated file again.

### Solutions

- Crash : Add verification to ask users permission to write files into external storage (/Downloads folder)
- double files : remove second download of generated certificate file and change its name:

From: `certificate.txt`
To: `wire-certificate-{user_handle}-{date-time}.txt`

#### How to Test

- Login to environment with MLS + E2EI enabled
- Generate certificate
- Go to : Settings -> Devices -> Current Device -> Certificate Details -> 3 dots at top-right corner -> download
- Success message should appear, app should NOT crash and when checking /Downloads folder it should contain only 1 certificate file

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -876,6 +876,8 @@
     <string name="media_gallery_on_image_download_error">There was an error when saving the image to
         Downloads folder. Please check your Internet connection and try again
     </string>
+    <string name="certificate_download_error">There was an error when saving the certificate to
+        Downloads folder. Please check your Internet connection and try again</string>
```
Comment (alexandreferris) [inline app/src/main/res/values/strings.xml]: Maybe we can change this string? As thinking a bit more on it, its used for when user denied permissions to write into external storage, any ideas welcome

**Comment by AndroidBob** [comment]
Build [3991](https://10.10.124.134/job/AR-build-pipeline/3991/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.6.3(9334416)(9334416)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.6.3(9334416)(9334416)-staging-compat.apk)

File: app/src/main/res/values/strings.xml (hunk 1 above)
Comment (MohamadJaara) [reply app/src/main/res/values/strings.xml]: for the copy please ask @Astrid-Wire

**Review by MohamadJaara — APPROVED** [review]

**Review by mchenani — APPROVED** [review]

File: app/src/main/res/values/strings.xml (hunk 1 above)
Comment (alexandreferris) [reply app/src/main/res/values/strings.xml]: This copy was removed due to not needing to ask for permissions anymore.

**Comment by AndroidBob** [comment]
Build [4002](https://10.10.124.134/job/AR-build-pipeline/4002/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.6.3(9335766)(9335766)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.6.3(9335766)(9335766)-staging-compat.apk)

**Pull request merged**